packages/ui/src/components/va-dropdown/VaDropdown.stories.ts

```diff
@@ -1,12 +1,386 @@
+        <template #anchor>
+          <div
+              style="width: 8rem; height: 8rem; border: var(--border-dashed)"
+              ref="cursorTarget"
+          >
+            Right click
+          </div>
+        </template>
+        Ctx dropdown
+      </va-dropdown>
+    `,
+})
+
+export const AnchorSelector = () => ({
+  components: { VaDropdown },
+  template: `
+      <div
+          style="border: var(--border-dashed)"
+          id="anchor-selector"
+      >
+        Click me
+      </div>
+      <va-dropdown
+          anchor-selector="#anchor-selector"
+      >
+        Anchor selector
+      </va-dropdown>
+    `,
+})
+
+export const InnerAnchorSelector = () => ({
   components: { VaDropdown },
-  template: '<VaDropdown/>',
+  template: `
+      <va-dropdown
+          inner-anchor-selector="#inner-anchor-selector"
+      >
+        <template #anchor>
+          <div style="border: var(--border-dashed)">
+            <div
+                style="margin: 1rem; border: var(--border-dashed)"
+                id="inner-anchor-selector"
+            >
+              Click me
+            </div>
+          </div>
+        </template>
+        Inner anchor selector
+      </va-dropdown>
+    `,
+})
+
+export const AnchorBoundProps = () => ({
+  components: { VaDropdown, VaIcon },
+  template: `
+      <va-dropdown>
+        <template #anchor="{ isOpened }">
+          <div>
+            <va-icon :name="isOpened ? 'va-arrow-up' : 'va-arrow-down'"/>
+          </div>
+        </template>
+        Clicked
+      </va-dropdown>
+    `,
+})
+
+export const KeepHeight = () => ({
+  components: { VaDropdown, VaButton, VaDropdownContent },
+  template: `
+      <div style="height: 1000px">
+        <va-dropdown>
+          <template #anchor>
+            <va-button>Anchor</va-button>
+          </template>
+          <va-dropdown-content>
+            <div style="height: 600px; width: 100px;">
+              Hi
+            </div>
+          </va-dropdown-content>
+        </va-dropdown>
+      </div>
+    `,
+})
+
+export const Autoplacement: StoryFn = () => ({
+  components: { VaDropdown, VaButton, VaDropdownContent, ScrollContainer },
+  directives: { scrollToMiddleY },
+  setup () {
+    return {
+      autoplacementTargetRef: ref(null),
+    }
+  },
+  template: `
+      Scroll vertically
+      <scroll-container
+          ref="autoplacementTargetRef"
+          v-scroll-to-middle-y
+      >
+        <div style="height: 200%;display: grid;place-items: center;">
+          <va-dropdown
+              :model-value="true"
+              :stateful="false"
+              :target="autoplacementTargetRef"
+          >
+            <template #anchor>
+              <div style="width: 6rem; height: 6rem; border: var(--border-dashed)">
+              </div>
+            </template>
+            Dropdown
+          </va-dropdown>
+        </div>
+      </scroll-container>
+    `,
+})
+
+export const AutoplacementWithHeight: StoryFn = () => ({
```

Comment:
```suggestion
export const AutoplacementWithScroll: StoryFn = () => ({
```

Probably could be both horizontal and vertical

Reply:
Left it verrical as horizontal scroll is showcased in story demonstrating sticking to edges behaviour